test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnector.cs

```
@@ -0,0 +1,166 @@
+            await ModuleUtil.ReportTestResultUntilSuccessAsync(
+                reportClient,
+                this.logger,
+                dmTestResult,
+                this.cancellationToken).ConfigureAwait(false);
+        }
+
+        async Task<Tuple<DateTime, HttpStatusCode>> SendDirectMethodAsync(
+            string deviceId,
+            string targetModuleId,
+            ModuleClient moduleClient,
+            string directMethodName,
+            DateTime runExpirationTime,
+            CancellationToken cancellationToken,
+            ILogger logger)
+        {
+            while ((!cancellationToken.IsCancellationRequested) && (DateTime.UtcNow < runExpirationTime))
+            {
+                try
+                {
+                    // Direct Method sequence number is always increasing regardless of sending result.
+                    Interlocked.Increment(ref this.directMethodCount);
+                    MethodRequest request = new MethodRequest(
+                        directMethodName,
+                        Encoding.UTF8.GetBytes($"{{ \"Message\": \"Hello\", \"DirectMethodCount\": \"{Interlocked.Read(ref this.directMethodCount).ToString()}\" }}"));
+                    MethodResponse result = await moduleClient.InvokeMethodAsync(deviceId, targetModuleId, request);
+                    if ((HttpStatusCode)result.Status == HttpStatusCode.OK)
+                    {
+                        logger.LogDebug(result.ResultAsJson);
+                    }
+                    else
+                    {
+                        logger.LogError(result.ResultAsJson);
+                    }
+
+                    logger.LogInformation($"[DirectMethodEdgeHubConnector] Invoke DirectMethod with count {Interlocked.Read(ref this.directMethodCount).ToString()}");
+                    return new Tuple<DateTime, HttpStatusCode>(DateTime.UtcNow, (HttpStatusCode)result.Status);
+                }
+                catch (IotHubCommunicationException e)
```

TODO: Swallow ALL exception